### PR TITLE
allow specification of entrypoint in the form of a slice

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
This matches the behaviour of buildah, and would allow a user to specify a multi option entrypoint.

File: cmd/podman/create.go [hunk 1]
```diff
@@ -555,16 +555,15 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
...
+	if c.IsSet("entrypoint") {
+		// ENTRYPOINT
+		if err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypoint); err != nil {
+			return nil, errors.Wrapf(err, "unable to translate entrypint %s", c.String("entrypoint"))
+		}
+	} else {
 		entrypoint = data.ContainerConfig.Entrypoint
 	}
-	// if entrypoint=, we need to clear the entrypoint
-	if len(entrypoint) == 1 && c.IsSet("entrypoint") && strings.Join(c.StringSlice("entrypoint"), "") == "" {
```
Comment (mheon) [inline cmd/podman/create.go]: We need to keep this logic - setting entrypoint explicitly to "" should clear entrypoint

File: cmd/podman/create.go [hunk 2]
```diff
@@ -555,16 +555,15 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
...
 
-	// ENTRYPOINT
-	// User input entrypoint takes priority over image entrypoint
-	entrypoint := c.StringSlice("entrypoint")
-	if len(entrypoint) == 0 && data != nil {
+	var entrypoint []string
+	if c.IsSet("entrypoint") {
+		// ENTRYPOINT
+		if err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypoint); err != nil {
+			return nil, errors.Wrapf(err, "unable to translate entrypint %s", c.String("entrypoint"))
```
Comment (TomSweeneyRedHat) [inline cmd/podman/create.go]: "entrypint" to "entrypoint"

Comment (mheon) [reply cmd/podman/create.go]: Does this work for single values? I tried unmarshalling a string into a []string in playground and it didn't work. We can't require all entrypoints to be arrays.

**Comment by rhatdan** [comment]
Finally Green.
@mheon @TomSweeneyRedHat @baude @umohnani8 @vrothberg @giuseppe PTAL

**Comment by mheon** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request closed without merging**